packages/iTwinUI-react/src/core/Select/Select.tsx

```diff
@@ -255,25 +309,54 @@ export const Select = <T,>(props: SelectProps<T>): JSX.Element => {
           isSelected,
           onClick: () => {
             !option.disabled && onChange?.(option.value);
-            close();
+            !multiple && close();
+          },
+          ref: (el: HTMLElement) => {
+            if (isSelected && !multiple) {
+              el?.scrollIntoView();
+            }
           },
-          ref: (el: HTMLElement) => isSelected && el?.scrollIntoView(),
           role: 'option',
           ...option,
           ...menuItem.props,
         });
       });
     },
-    [itemRenderer, onChange, options, value],
+    [itemRenderer, multiple, onChange, options, value],
   );
 
-  const selectedItem = React.useMemo(() => {
+  const selectedItems = React.useMemo(() => {
     if (value == null) {
       return undefined;
     }
-    return options.find((option) => option.value === value);
+    return Array.isArray(value)
+      ? options.filter((option) => value.some((val) => val === option.value))
+      : options.find((option) => option.value === value);
   }, [options, value]);
 
+  const selectedItemsArray = React.useMemo(() => {
+    if (!selectedItems || !Array.isArray(selectedItems)) {
```

Comment:
Why not checking whther `multiple` is enabled`

Reply:
Cause I am trusting the props guard, which should not allow to pass array of values, when disabled `multiple`